src/FileConventions/Library.fs

```diff
@@ -104,3 +103,66 @@ let EolAtEof(fileInfo: FileInfo) =
                 False
         else
             True
+
+let DetectInconsistentVersionsInGitHubCIWorkflow(fileInfos: seq<FileInfo>) =
+    fileInfos
+    |> Seq.iter(fun fileInfo -> assert (fileInfo.FullName.EndsWith ".yml"))
+
+    let inconsistentPulumiVersions =
+        fileInfos
+        |> Seq.map(fun fileInfo -> File.ReadLines fileInfo.FullName)
+        |> Seq.map(fun fileLines ->
+            fileLines
+            |> Seq.filter(fun line -> line.Contains "pulumi-version:")
+            |> Seq.map(fun line -> line.Substring(line.IndexOf(":") + 1))
+            |> Seq.map(fun line -> line.Trim())
+        )
+        |> Seq.concat
+        |> Set.ofSeq
+        |> Seq.length
+        |> (fun length -> length > 1)
+
+    let versionRegex = Regex("\\s([^\\s]*)@([^\\s]*)\\s", RegexOptions.Compiled)
+
+    let mutable versionMap: Map<string,Set<string>> = Map<string, Set<string>> []
+
+    let addSet (value: string) (prevSet: Set<string> option) = 
+        match prevSet with
+            | Some pSet -> Some (Set.add value pSet)
+            | None -> None
+        
+    fileInfos
+    |> Seq.iter(fun fileInfo ->
+        let fileText = File.ReadAllText fileInfo.FullName
+
+        (versionRegex.Matches fileText)
+        |> Seq.iter (fun regexMatch -> 
+                let key = regexMatch.Groups.[1].ToString()
+                let value = regexMatch.Groups.[2].ToString()
+                if versionMap.ContainsKey(key) then
+                    versionMap <- versionMap.Change(key, addSet value)
+                else
+                    versionMap <- versionMap.Add(key, Set.singleton(value))
```

Comment:
@realmarv no need for parens in the singleton call

Reply:
Done